trap_posix.go

```diff
@@ -11,13 +11,22 @@ import (
 
 func (d *Daemon) trapSignals() {
 	ch := make(chan os.Signal, 1)
-	signal.Notify(ch, syscall.SIGUSR1)
+	signal.Notify(ch, syscall.SIGUSR1, syscall.SIGINT, syscall.SIGTERM, syscall.SIGABRT)
 	for {
 		select {
 		case s := <-ch:
 			switch s {
 			case syscall.SIGUSR1:
 				d.handleSIGUSR1()
+			case syscall.SIGINT:
```

Comment:
Opportunity to simplify:

```suggestion
			case syscall.SIGINT, syscall.SIGTERM, syscall.SIGABRT:
				d.Close()
				os.Exit(int(s))
```

Reply:
`s` is an interface so can't be cast to `int`, but take a look at my update and see if it is preferable